File: split/stackintobox.pas
procedure stackintobox(b: halfword; f: internalfontnumber; c: quarterword);
var
  p: halfword;
begin
  p := charbox(f, c);
  mem[p].hh.rh := mem[b + 5].hh.rh;
  mem[b + 5].hh.rh := p;
  mem[b + 3].int := mem[p + 3].int;
end;

File: cpp/stackintobox.cpp
#include "stackintobox.h"
#include "charbox.h"

void stackintobox(halfword b, internalfontnumber f, quarterword c)
{
	auto p = charbox(f, c);
	link(p) = list_ptr(b);
	list_ptr(b) = p;
	height(b) = height(p);
}
